tests/Silverback.Integration.Tests.E2E/Kafka/ErrorHandlingTests.cs

```diff
@@ -1177,5 +1265,101 @@ public async Task MovePolicy_ToOtherTopicAfterRetry_MessageRetriedAndMoved()
+                                    config =>
+                                    {
+                                        config.BootstrapServers = "PLAINTEXT://e2e";
+                                    })
+                                .AddOutbound<IIntegrationEvent>(
+                                    endpoint => endpoint.ProduceTo(DefaultTopicName))
+                                .AddInbound(
+                                    endpoint => endpoint
+                                        .ConsumeFrom(DefaultTopicName)
+                                        .OnError(
+                                            policy => policy.MoveToKafkaTopic(
+                                                moveEndpoint => moveEndpoint.ProduceTo("other-topic")))
+                                        .Configure(
+                                            config =>
+                                            {
+                                                config.GroupId = "consumer1";
+                                            })))
+                        .AddIntegrationSpy()
+                        .AddDelegateSubscriber(
+                            (IIntegrationEvent _) => throw new InvalidOperationException("Move!")))
+                .Run();
+
+            var publisher = Host.ScopedServiceProvider.GetRequiredService<IEventPublisher>();
+            await publisher.PublishAsync(message1);
+            await publisher.PublishAsync(message2);
+            await publisher.PublishAsync(message3);
+            await Helper.WaitUntilAllMessagesAreConsumedAsync();
+
+            Helper.Spy.OutboundEnvelopes.Should().HaveCount(6);
+            Helper.Spy.InboundEnvelopes.Should().HaveCount(3);
+
+            var otherTopic = Helper.GetTopic("other-topic");
+            var otherTopicMessages = otherTopic.GetAllMessages();
+            otherTopic.MessagesCount.Should().Be(3);
+            otherTopicMessages[0].Value.Should()
+                .BeEquivalentTo(Helper.Spy.InboundEnvelopes[0].RawMessage.ReReadAll());
+            otherTopicMessages[1].Value.Should()
+                .BeEquivalentTo(Helper.Spy.InboundEnvelopes[1].RawMessage.ReReadAll());
+            otherTopicMessages[2].Value.Should()
+                .BeEquivalentTo(Helper.Spy.InboundEnvelopes[2].RawMessage.ReReadAll());
+
+            Helper.Spy.OutboundEnvelopes[5].Endpoint.Name.Should().Be("other-topic");
+            Helper.Spy.OutboundEnvelopes[5].Headers
+                .Should().ContainEquivalentOf(
+                    new MessageHeader(
+                        DefaultMessageHeaders.FailedAttempts,
+                        1));
+            Helper.Spy.OutboundEnvelopes[5].Headers
+                .Should().ContainEquivalentOf(
+                    new MessageHeader(
+                        DefaultMessageHeaders.FailureReason,
+                        "System.Reflection.TargetInvocationException in System.Private.CoreLib"));
```

Comment:
Really? This is what we expect and what we will settle for?

Reply:
You're right, but it's what I find in the execution of e2e.. uhm...